osu.Game/Screens/Import/FileImportScreen.cs

```diff
@@ -131,10 +148,37 @@ public override bool OnExiting(ScreenExitEvent e)
             return base.OnExiting(e);
         }
 
-        private void directoryChanged(ValueChangedEvent<DirectoryInfo> _)
+        private void directoryChanged(ValueChangedEvent<DirectoryInfo> directoryChangedEvent)
         {
             // this should probably be done by the selector itself, but let's do it here for now.
             fileSelector.CurrentFile.Value = null;
+
+            // enable the "importDirectoryButton" only when there is at least 1 file that matches the extension
+            importAllButton.Enabled.Value = false;
+
+            if (directoryChangedEvent.NewValue == null)
+                return;
+
+            DirectoryInfo directoryInfo = directoryChangedEvent.NewValue;
+
+            if (!directoryInfo.Exists)
+                return;
+
+            try
+            {
+                foreach (FileInfo file in directoryInfo.EnumerateFiles())
+                {
+                    if (game.HandledExtensions.Contains(file.Extension))
+                    {
+                        importAllButton.Enabled.Value = true;
+                        break;
+                    }
+                }
```

Comment:
I would have probably LINQ'd this, I don't see a world where this is performance-critical, but I guess it's fine